src/test/java/uk/gov/pay/directdebit/mandate/dao/SearchMandateDaoIT.java

```diff
@@ -53,59 +59,84 @@ public void setup() {
     }
     
     @Test
     @Parameters({"STATEMENT123", "statement123", "ment"})
     public void searchByBankStatementReference(String searchString) {
-        var mandateSearchParams = aMandateSearchParams()
+        var searchParams = aMandateSearchParams()
                 .withMandateBankStatementReference(MandateBankStatementReference.valueOf(searchString))
                 .withGatewayAccountId("gateway-account-id");
-        List<Mandate> results = mandateSearchDao.search(mandateSearchParams);
+        List<Mandate> results = mandateSearchDao.search(searchParams);
         assertThat(results.size()).isEqualTo(1);
-        assertThat(results.get(0)).isEqualTo(mandate2.toEntity());
+        assertThat(results.get(0).getId()).isEqualTo(mandate2.getId());
     }
     
     @Test
     @Parameters({"joe.bloggs@example.com", "joe.bloggs@EXAMPLE.com", "joe.bloggs"})
     public void searchByEmail(String searchString) {
-        var mandateSearchParams = aMandateSearchParams().withEmail(searchString).withGatewayAccountId("gateway-account-id");
-        List<Mandate> results = mandateSearchDao.search(mandateSearchParams);
+        var searchParams = aMandateSearchParams().withEmail(searchString).withGatewayAccountId("gateway-account-id");
+        List<Mandate> results = mandateSearchDao.search(searchParams);
         assertThat(results.size()).isEqualTo(1);
-        assertThat(results.get(0)).isEqualTo(mandate1.toEntity());
+        assertThat(results.get(0).getId()).isEqualTo(mandate1.getId());
     }
 
     @Test
     @Parameters({"JOe Bloggs", "joe bloggs", "bloggs"})
     public void searchByName(String searchString) {
-        var mandateSearchParams = aMandateSearchParams().withName(searchString).withGatewayAccountId("gateway-account-id");
-        List<Mandate> results = mandateSearchDao.search(mandateSearchParams);
+        var searchParams = aMandateSearchParams().withName(searchString).withGatewayAccountId("gateway-account-id");
+        List<Mandate> results = mandateSearchDao.search(searchParams);
         assertThat(results.size()).isEqualTo(1);
-        assertThat(results.get(0)).isEqualTo(mandate1.toEntity());
+        assertThat(results.get(0).getId()).isEqualTo(mandate1.getId());
     }
     
     @Test
     public void searchByFromDate() {
-        
+        var searchParams = aMandateSearchParams().withFromDate(now().minusHours(1)).withGatewayAccountId("gateway-account-id");
+        List<Mandate> results = mandateSearchDao.search(searchParams);
+        assertThat(results.size()).isEqualTo(1);
+        assertThat(results.get(0).getId()).isEqualTo(mandate1.getId());
+
+        searchParams = aMandateSearchParams().withFromDate(now().minusHours(7)).withGatewayAccountId("gateway-account-id");
+        results = mandateSearchDao.search(searchParams);
+        assertThat(results.size()).isEqualTo(2);
+        assertThat(results.stream().map(m -> m.getId()).collect(toList())).contains(mandate1.getId(), mandate2.getId());
```

Comment:
`assertThat(results.stream().map(Mandate::getId).collect(toSet()), is(List.of(mandate1.getId(), mandate2.getId()))` might be clearer and will also check the size.